app/lib/actions/pulp/abstract_async_task_group.rb

```diff
@@ -0,0 +1,81 @@
+module Actions
+  module Pulp
+    class AbstractAsyncTaskGroup < Pulp::Abstract
+      include Actions::Base::Polling
+
+      # A call report (documented https://github.com/pulp/pulp/blob/master/docs/dev-guide/integration/rest-api/consumer/applicability.rst#id65)
+      # Looks like:  {
+      #     "_href": "/pulp/api/v2/task_groups/7744e2df-39b9-46f0-bb10-feffa2f7014b/",
+      #     "group_id": "7744e2df-39b9-46f0-bb10-feffa2f7014b"
+      #     }
+      #
+      #
+
+      # A TaskGroup (https://github.com/pulp/pulp/blob/master/docs/dev-guide/integration/rest-api/tasks.rst#task-group-management)
+      # Looks like:
+      # {
+      #  "accepted": 0,
+      #  "finished": 100,
+      #  "running": 4,
+      #  "canceled": 0,
+      #  "waiting": 2,
+      #  "skipped": 0,
+      #  "suspended": 0,
+      #  "error": 0,
+      #  "total": 106
+      # }
+
+      def run(event = nil)
+        # do nothing when the action is being skipped
+        unless event == Dynflow::Action::Skip
+          super
+        end
+      end
+
+      def humanized_state
+        case state
+        when :running
+          if self.external_task.nil?
+            _("initiating Pulp task")
+          else
+            _("checking Pulp task status")
+          end
+        else
+          super
+        end
+      end
+
+      def done?
+        finishing_states = ["finished", "canceled", "skipped", "suspended", "error"]
+        return false if (finishing_states - external_task.keys).present?
+        sum = 0
+        ["finished", "canceled", "skipped", "suspended", "error"].each do |state|
+          sum += external_task[state]
+        end
+        sum == external_task["total"]
```

Comment:
Good point .. I am actually tempted to remove that "completed?" method from runcible. This is because the way the dynflow wants things. It wants a "done"   method that doesn't do a remote call but rather assesses the summary retrieved from poll_external task. While the "completed" method in runcible does a remote call to fetch the summary. If you are ok with that, I am going remove the "completed?" in that runcible patch